jsonnet/kube-prometheus/kube-prometheus-custom-metrics.libsonnet

```diff
@@ -0,0 +1,141 @@
+local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
+
+// Custom metrics API allows the HPA v2 to scale based on arbirary metrics.
+// For more details on usage visit https://github.com/DirectXMan12/k8s-prometheus-adapter#quick-links
+
+{
+  _config+:: {
+    prometheusAdapter+:: {
+      // Rules for custom-metrics
+      config: |||
+        rules:
```

Comment:
to avoid duplication, how about we make this config not an opaque string, but rather structured, so we can just merge the additonal configuration in, instead of having to re-specify it, which is prone to error when we fix things in one place but not the other

Reply:
This would also mean adjusting https://github.com/coreos/kube-prometheus/blob/master/jsonnet/kube-prometheus/prometheus-adapter/prometheus-adapter.libsonnet#L20, correct?

Happy to give it a try :-)

Reply:
correct

Reply:
Made the change of the config in https://github.com/coreos/kube-prometheus/pull/485, will update this PR once this looks good :-)

Reply:
Merged that :)

Reply:
@brancz updated :-)